src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala

```diff
@@ -0,0 +1,176 @@
+import org.apache.spark.sql.functions.{explode, from_unixtime, to_timestamp, udf}
+import org.locationtech.geomesa.spark.jts.{udf => _, _}
+import org.locationtech.jts.{geom => jts}
+import org.locationtech.jts.geom.prep._
+import osmesa.analytics.Analytics
+import vectorpipe.sources.Source
+
+import java.net.URI
+
+// These only required for test getAreaIndex implementation
+import geotrellis.vector.io._
+import geotrellis.vector.io.json._
+import spray.json._
+
+object StreamingAOIMonitor extends CommandApp(
+  name = "streaming-aoi-monitor",
+  header = "Streaming AOI Monitor",
+  main = {
+    val intervalOpt =
+      Opts
+        .option[String]("interval", help = "Period of time to aggregate over (d=daily, w=weekly, m=monthly)")
+        .validate("Must be one of 'd', 'w', or 'm'") { arg: String => arg.length == 1 && "dwm".contains(arg.apply(0)) }
+
+    val databaseUrlOpt =
+      Opts
+        .option[URI](
+        "database-url",
+        short = "d",
+        metavar = "database URL",
+        help = "URL of database containing AOI table (default: DATABASE_URL environment variable)"
+      )
+      .orElse(Opts.env[URI]("DATABASE_URL", help = "The URL of the database"))
+
+    val augmentedDiffSourceOpt =
+      Opts
+        .option[URI](
+        "augmented-diff-source",
+        short = "a",
+        metavar = "uri",
+        help = "Location of augmented diffs to process"
+      )
+
+    val startSequenceOpt =
+      Opts
+        .option[Int](
+        "start-sequence",
+        short = "s",
+        metavar = "sequence",
+        help = "Starting sequence. If absent, the current (remote) sequence will be used."
+      )
+      .orNone
+
+    val endSequenceOpt =
+      Opts
+        .option[Int](
+        "end-sequence",
+        short = "e",
+        metavar = "sequence",
+        help = "Ending sequence. If absent, this will be an infinite stream."
+      )
+      .orNone
+
+    (intervalOpt, databaseUrlOpt, augmentedDiffSourceOpt, startSequenceOpt, endSequenceOpt).mapN {
+      (interval, databaseUri, augmentedDiffSource, startSequence, endSequence) =>
+      val appName = "StreamingAOIMonitor"
+
+      implicit val spark: SparkSession = Analytics.sparkSession("Streaming AOI Monitor")
+
+      import spark.implicits._
+      spark.withJTS
+      import AOIMonitorUtils._
+
+      val aoiIndex = getAreaIndex(/*databaseUri*/)
+
+      val options = Map(
+        Source.BaseURI -> augmentedDiffSource.toString,
+        Source.ProcessName -> appName
+      ) ++
+      startSequence
+        .map(s => Map(Source.StartSequence -> s.toString))
+        .getOrElse(Map.empty[String, String]) ++
+      endSequence
+        .map(s => Map(Source.EndSequence -> s.toString))
+        .getOrElse(Map.empty[String, String])
+
+      val diffs = spark
+        .readStream
+        .format(Source.AugmentedDiffs)
+        .options(options)
+        .load
+        .withColumn("watermark", to_timestamp(from_unixtime('sequence * 60 + 1347432900)))
```

Comment:
What are the values of the constants here? It would be helpful if these were named to more easily understand the code's intent.

Reply:
👍